docs/changelog2.rst

```diff
@@ -0,0 +1,241 @@
+Changelog
+=========
+
+
+Version 2.0.0 (2022-??-??)
+--------------------------
+
+Refactor of the internals
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+No longer using ctypes, but C extension
+
+Top-level element-wise functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Not only all methods on the Geometry classes, but also some methods from submodules (eg ..)
+
+* Vectorized constructor functions
+* Optionally output to a user-specified array (``out`` keyword argument) when constructing
+  geometries from ``indices`` (#380).
+* Enable bulk construction of geometries with different number of coordinates
+  by optionally taking index arrays in all creation functions (#230, #322, #326, #346).
+
+
+API changes (deprecated in 1.8)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+See :ref:`migration` for more details on how to update your code.
+
+Recommended to first upgrade to Shapely 1.8 and resolve all deprecation
+warnings before upgrading to Shapely 2.0.
+
+Summary of changes:
+
+* Geometries are now immutable and hashable.
+* Multi-part geometries no longer behave as "sequences". This means that they
+  no longer have a length, are not iterable and indexable anymore. Use ``.geoms``
+  attribute instead to access the parts.
+* Geometries no longer directly implement the numpy array interface. To
+  convert to an array of coordinates, use `.coords` instead
+  (`np.asarray(geom.coords)`).
+* Consistent creation of empty geometries (for exampe ``Polygon()`` now
+  actually creates an empty Polygon instead of an empty geometry collection).
+* The following attributes and methods on the Geometry classes were
+  deprecated and are now removed:
+  * ``array_interface()`` and ``ctypes``
+  * ``asShape()``, and the adapters classes to create geometry-like proxy
+    objects (use ``shape()`` instead).
+  * ``empty()`` method
+
+Breaking API changes
+^^^^^^^^^^^^^^^^^^^^
+
+Some additional backwards incompatible API changes were included in Shapely
+2.0 that were not yet deprecated in Shapely 1.8:
+
+* The default of the ``preserve_topology`` keyword of ``simplify()`` changed
+  to True (#1392).
+* A ``GeometryCollection`` that consists of all empty sub-geometries now
+  returns those empty geometries in ``.geoms`` (instead of returning an empty
+  list) (#1420).
+* The unused ``shape_factory()`` method and ``HeterogeneousGeometrySequence``
+  class are removed (#1421).
+* The undocumted ``__geom__`` attribute is removed. To access the raw GEOS pointer,
+  the ``_geom`` attribute is still present (1417).
+
+New features
+^^^^^^^^^^^^
+
+More informative repr with (truncated) WKT
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+Support for fixed precision model for geometries and in overlay functions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* `set_precision` to conform a geometry to a certain grid size, and this will then also be used by subsequent overlay methods
+* `grid_size` keyword in overlay methods (`intersection`, `union`, `difference`, etc) 
```

Comment:
This also reminds me that we should actually expose this keyword in the Geometry class methods as well

Reply:
-> https://github.com/shapely/shapely/pull/1468